zustand/useHighscoresStore.js

```diff
@@ -4,15 +4,38 @@ import { nanoid } from "nanoid";
 import { useState, useEffect } from "react";
 
 const useHighscoresStore = createLocalStorageStore(
-  (set) => ({
+  (set, get) => ({
     highscores: [],
-    addHighscore: (name, time, failed) =>
+    addHighscore: (name, time, formattedTime, failed) => {
       set((state) => ({
         highscores: [
           ...state.highscores,
-          { name: name, time: time, failed: failed, id: nanoid() },
+          {
+            name: name,
+            time: time,
+            formattedTime: formattedTime,
+            failed: failed,
+            id: nanoid(),
+          },
         ],
-      })),
+      }));
+      get().sortByTime();
+      get().sortByFailed();
+    },
+    sortByFailed: () => {
+      set((state) => {
+        return {
+          highscores: state.highscores.sort((a, b) => a.failed - b.failed),
+        };
+      });
+    },
+    sortByTime: () => {
+      set((state) => {
+        return {
+          highscores: state.highscores.sort((a, b) => a.time - b.time),
+        };
+      });
+    },
```

Comment:
please turn this into a single sort function. I think you'll never have the use case that you want need to sort by only one criteria.

You can check out this article on how to to that:
https://gomakethings.com/sorting-an-array-by-multiple-criteria-with-vanilla-javascript/

Also i'd suggest not sorting in separate methods since that is a bit overcomplicated. You can just sort in place at the time of creation of the new array after adding. This way you also do not need the get function.

See my revised store here:

```js
const useHighscoresStore = createLocalStorageStore(
  (set) => ({
    highscores: [],
    addHighscore: (name, time, formattedTime, failed) => {
      set((state) => ({
        highscores: [
          ...state.highscores,
          {
            name,
            time,
            formattedTime,
            failed,
            id: nanoid(),
          },
        ].sort((a, b) => {
          // sort by failed first
          if (a.failed < b.failed) return -1;
          if (a.failed > b.failed) return 1;

          // then sort by time
          if (a.time < b.time) return -1;
          if (a.time > b.time) return 1;
          
          return 0;
        }),
      }));
    },
  }),
  "higscores"
);
```

Reply:
I splitted this up because in the next userstory I want the user to choose the way of sorting. you can sort by `Time` or by `FailedAttempts` later.